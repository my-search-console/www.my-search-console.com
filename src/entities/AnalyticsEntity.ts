type Base = {
  label?: string
  value?: string
  data?: { [x: string]: string | number | boolean }
}

type PagesEvents = {
  category: "pages"
  action: "search"
  data: {
    query: string
  }
}

export type AuthenticationAnalyticsEntity = Base & {
  category: "authentication"
  action: "logout" | "login"
}

export type AnalyticsPaymentEntityPaymentSources =
  | "query"
  | "country"
  | "device"
  | "source"
  | "date"
  | "indexation/filter"
  | "indexation/save"
  | "indexation/index-all"
  | "indexation/filter-indexing"
  | "indexation/quota"
  | "indexation/trial-expired"
  | "indexation/add-website"
  | "indexation/auto-index"
  | "landing"
  | "indexation/report"
  | "navbar"

export type ModalAnalyticsEntity = Base &
  (
    | {
        category: "modal"
        action: "close"
      }
    | {
        category: "modal"
        action: "open"
        data: {
          type: "payment"
          source: AnalyticsPaymentEntityPaymentSources
        }
      }
  )

export type NewsAnalyticsEntity = Base &
  (
    | {
        category: "news"
        action: "close"
      }
    | {
        category: "news"
        action: "open"
      }
  )

export type ShowoffAnalyticsEntity = Base & {
  category: "show-off"
  action: "download" | "fetch"
}

export type RoastAnalyticsEntity = Base & {
  category: "roast"
  action: "refresh-all"
}

export type PaymentAnalyticsEntity = Base & {
  category: "payment"
  action: "open" | "success" | "close"
}

export type WebsiteAnalyticsEntity = Base & {
  category: "websites"
  action: "create" | "delete"
}

export type ErrorAnalyticsEntity = Base & {
  category: "error"
  action: "login"
  message: string
}

export type SiteSwitcherAnalyticsEntity = Base & {
  category: "site_switcher"
  action: "switch_site"
}

export type IndexationAnalyticsEntity = Base & {
  category: "indexation"
  action: "index"
  data: {
    google: boolean
    yandex: boolean
    bing: boolean
    naver: boolean
  }
}

export type AnalyticsEntity =
  | PagesEvents
  | RoastAnalyticsEntity
  | NewsAnalyticsEntity
  | WebsiteAnalyticsEntity
  | AuthenticationAnalyticsEntity
  | PaymentAnalyticsEntity
  | ErrorAnalyticsEntity
  | IndexationAnalyticsEntity
  | ModalAnalyticsEntity
  | ShowoffAnalyticsEntity
