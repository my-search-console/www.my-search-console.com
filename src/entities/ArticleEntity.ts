export enum ArticleTypes {
  QUOTE = "article/quote",
  RICH_TEXT = "article/rich_text",
  TITLE = "article/title",
  IMAGE = "article/image",
  TOC = "article/toc",
  VIDEO = "article/video",
  RECIPE = "article/recipe",
  PRODUCT = "article/product",
  ARTICLE = "article/article",
}

export type ArticleTOC = {
  id: string
  type: ArticleTypes.TOC
  title: {
    component: string
    value: string
  }
  items: Array<{ label: string; to: string; depth: number }>
}

export type ArticleVideo = {
  id: string
  type: ArticleTypes.VIDEO
  src: any
  ring: string
  illustration: any
  alt: string
  autoplay: boolean
}

export type ArticleProduct = {
  id: string
  type: ArticleTypes.PRODUCT
  product: {
    buttons: { url: string; label: string; theme: "primary" | "secondary" }[]
    name: string
    lang: string
    description: string
    image: {
      src: any
      alt: string
    }
  }
}

export type ArticleRecipe = {
  id: string
  type: ArticleTypes.RECIPE
  difficulty: string
  preparationDuration: number
  cookDuration: number
  recipeCuisine: string
  recipeCategory: string
  calories: number
  name: string
  description: string
  keywords: string
  image169: {
    src: any
  }
  image43: {
    src: any
  }
  image11: {
    src: any
  }
  tools: Array<{ quantity: number; url: string | null; value: string }>
  ingredients: {
    portions: number
    items: Array<{
      quantity: number
      url: string | null
      value: string
      weight: string
    }>
  }
  steps: Array<{
    content: string
    name: string
    illustration: {
      alt: string
      src: any | null
    }
  }>
}

export type ArticleQuote = {
  id: string
  type: ArticleTypes.QUOTE
  text: string
  author: string
}

export type ArticleRichText = {
  id: string
  type: ArticleTypes.RICH_TEXT
  content: string
}

export type ArticleTitle = {
  type: ArticleTypes.TITLE
  value: string
  id: string
  component: string
  faq: boolean
}

export type ArticleImage = {
  id: string
  type: ArticleTypes.IMAGE
  src?: string
  alt: string
  legend: string
}

export type ArticleToArticle = {
  id: string
  type: ArticleTypes.ARTICLE
  article: ArticleEntity
}

export type ArticleEntity = {
  id: string
  title: string
  lang: string
  hidden: boolean
  version: "v1" | "v2" | null
  canonical: string
  description: string
  published_at: Date
  updated_at: Date
  url: string
  meta: {
    title: string
    description: string
    indexable: boolean
  }
  illustration: {
    alt: string
    src: any
  }
  preview: {
    alt: string
    src: any
  }
  content: Array<
    | ArticleQuote
    | ArticleRichText
    | ArticleTitle
    | ArticleImage
    | ArticleTOC
    | ArticleRecipe
    | ArticleProduct
    | ArticleToArticle
    | ArticleVideo
  >
  author: {
    name: string
    id: string
    bio: string
    lang: string
    url: string
    staff: boolean
    username: string
    image: {
      alt: string
      src: any
    }
  }
  category: {
    id: string
    meta: {
      title: string
      description: string
    }
    title: string
    description: string
    url: string
    name: string
    hidden: boolean
    lang: string
  }
  relatedArticles: ArticleEntity[]
}
