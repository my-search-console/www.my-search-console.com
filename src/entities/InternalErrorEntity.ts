export enum InternalErrorEntity {
  GOOGLE_BROWSER_CLOSED = "google/browser-closed",
  INDEXATION_AUTO_ACTIVATED_WITHOUT_SEARCH_ENGINE = "notifications/indexation/auto/activated-without-search-engines",
}
