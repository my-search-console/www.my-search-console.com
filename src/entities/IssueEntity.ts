export enum IssueTypes {
  robots = "robots",
  server = "server",
  sitemap = "sitemap",
}

export enum IssueNames {
  "robots/crawl-allowed" = "robots/crawl-allowed",
  "robots/empty" = "robots/empty",
  "robots/javascript" = "robots/javascript",

  "server/error" = "server/error",
  "server/not-https" = "server/not-https",
  "server/not-sc-domain/not-same-url" = "server/not-sc-domain/not-same-url",
  "server/not-sc-domain" = "server/not-sc-domain",
  "server/no-index" = "server/no-index",

  "sitemap/no-sitemap" = "sitemap/no-sitemap",
  "sitemap/server-error" = "sitemap/server-error",
  "sitemap/bad-formed" = "sitemap/bad-formed",
  /**
   * Quand il y a une entr√©e hreflang mais sans la correspondance
   */
  "sitemap/hreflang-without-loc" = "sitemap/hreflang-without-loc",
  "sitemap/contains-external-url" = "sitemap/contains-external-url",
  "sitemap/no-same-website" = "sitemap/no-same-website",
}

export type IssueEntity = {
  id: string
  fk_website_id: string
  type: IssueTypes
  name: IssueNames
  status: "pass" | "dont-pass" | "warning"
  context: any
  created_at: Date
  updated_at: Date
}
