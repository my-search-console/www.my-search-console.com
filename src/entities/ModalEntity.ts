export enum ModalKeys {
  "upsell-confirmation-modal" = "upsell-confirmation-modal",
  "remove-website-confirmation-modal" = "remove-website-confirmation-modal",
  "logs-create-modal" = "logs-create-modal",
  "add-user-to-website-modal" = "add-user-to-website-modal",
  "indexation-auto-modal" = "indexation-auto-modal",
}
