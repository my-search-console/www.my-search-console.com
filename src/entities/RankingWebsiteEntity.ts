export interface RankingStatEntity {
  id: string
  fk_website_id: string
  date: Date | string
  clicks: number
  impressions: number
  position: number
  click_through_rate: number
  previous_clicks: number
  previous_impressions: number
  previous_position: number
  previous_click_through_rate: number
  query: string
  device: "desktop" | "mobile" | "tablet" | "unknown"
  country: string
  source: "google" | "bing" | "yandex"
  page: string
}

export type RankingOrderByType =
  | "clicks"
  | "impressions"
  | "position"
  | "click_through_rate"

export interface RankingStatsForFrontend {
  global: Pick<
    RankingStatEntity,
    | "click_through_rate"
    | "clicks"
    | "impressions"
    | "position"
    | "previous_clicks"
    | "previous_click_through_rate"
    | "previous_impressions"
    | "previous_position"
  >
  date: Array<
    Pick<
      RankingStatEntity,
      | "date"
      | "click_through_rate"
      | "clicks"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
  sources: Array<
    Pick<
      RankingStatEntity,
      | "click_through_rate"
      | "clicks"
      | "source"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
  devices: Array<
    Pick<
      RankingStatEntity,
      | "click_through_rate"
      | "clicks"
      | "device"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
  pages: Array<
    Pick<
      RankingStatEntity,
      | "click_through_rate"
      | "clicks"
      | "page"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
  query: Array<
    Pick<
      RankingStatEntity,
      | "click_through_rate"
      | "clicks"
      | "query"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
  countries: Array<
    Pick<
      RankingStatEntity,
      | "click_through_rate"
      | "clicks"
      | "country"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
  google: Array<
    Pick<
      RankingStatEntity,
      | "date"
      | "click_through_rate"
      | "clicks"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
  yandex: Array<
    Pick<
      RankingStatEntity,
      | "date"
      | "click_through_rate"
      | "clicks"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
  bing: Array<
    Pick<
      RankingStatEntity,
      | "date"
      | "click_through_rate"
      | "clicks"
      | "impressions"
      | "position"
      | "previous_clicks"
      | "previous_click_through_rate"
      | "previous_impressions"
      | "previous_position"
    >
  >
}

export interface LeaderboardWebsitesEntity {
  id: string
  clicks: number
  impressions: number
}
