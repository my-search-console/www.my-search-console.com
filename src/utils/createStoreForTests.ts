import { init as createStore } from "../redux/store"
import { actions } from "../redux/actions"
import { TestModule } from "../modules/TestModule"

export const createStoreForTests = (initialState = {}) => {
  const di = new TestModule().build()

  const { store } = createStore(initialState, di)

  return {
    store,
    actions,
    di,
  }
}
