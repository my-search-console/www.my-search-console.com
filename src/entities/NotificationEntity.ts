import { ErrorEntity } from "@my-search-console/interfaces"
import { InternalErrorEntity } from "./InternalErrorEntity"

export enum NotificationMessageEntity {
  WEBSITES_CREATE_SUCCESS = "websites/success/create",
  WEBSITES_ALREADY_REFRESHING = "websites/already-refreshing",
  FOUDROYER_PAYMENT_SUCCESS = "payment/success",
  BUILDER_PAYMENT_SUCCESS = "builder/payment/create",
  SYNC_SUCCESS = "websites/success/sync",
  WEBSITES_UPDATE_SUCCESS = "websites/success/update",
  WEBSITES_SITEMAP_UPDATE_SUCCESS = "websites/sitemap/update/empty",
  WEBSITES_NOT_SELECTED = "websites/not-selected",
  WEBSITES_SITEMAP_UPDATE_EMPTY = "websites/sitemap/update/empty",
  WEBSITES_CREDENTIALS_UPDATE_EMPTY = "websites/credentials/update/empty",
  INDEXATION_SUCCESS = "indexation/success",
  INDEXATION_NO_FILTER_SELECTED = "indexation/notifications/no-filter-selected",
  KEYWORDS_ADD_SUCCESS = "keywords/add/success",
  WEBSITES_DELETED = "notifications/website-removed",
}

export type NotificationEntity = {
  type: "info" | "success" | "error" | "warning"
  message: ErrorEntity | InternalErrorEntity | NotificationMessageEntity
  id: number
  timeout?: number
  onValidate?: Function
}
