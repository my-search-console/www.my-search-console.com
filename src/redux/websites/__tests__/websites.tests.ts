import { ErrorEntity } from "@my-search-console/interfaces"
import { NotificationMessageEntity } from "../../../entities/NotificationEntity"
import {
  WebsiteActivated,
  WebsiteNoSitemap,
  WebsiteNotActivated,
} from "../../../modules/seeds/WebsitesSeeds"
import { createStoreForTests } from "../../../utils/createStoreForTests"

describe("websites tests suite", () => {
  describe("creation d'un site web", () => {
    it("en tant qu'utilisateur, je dois pouvoir afficher la modale de création de site", async () => {
      const { store, actions, di } = createStoreForTests()

      expect(store.getState().websites.createWebsiteModal.isOpen).toEqual(false)

      await store.dispatch<any>(actions.websites.$WebsiteCreateModal())

      expect(store.getState().websites.createWebsiteModal.isOpen).toEqual(true)
    })

    it("en tant qu'utilisateur, quand j'ouvre la modale de création, je dois pouvoir sélectionner un site google search console", async () => {
      const { store, actions, di } = createStoreForTests()

      const domains = [{ id: "www.sudoku.academy" }]

      di.WebsitesRepository.__storeFetchGoogleDomainsResponse(domains)

      await store.dispatch<any>(actions.websites.$WebsiteCreateModal())

      expect(store.getState().websites.createWebsiteModal.type).toEqual(
        "google"
      )
      expect(store.getState().websites.createWebsiteModal.domains).toEqual(
        domains
      )

      store.dispatch<any>(
        actions.websites.WebsiteAddSourceModalSelect({ value: domains[0].id })
      )

      expect(store.getState().websites.createWebsiteModal.selected).toEqual(
        domains[0].id
      )
    })

    it("en tant qu'utilisateur, si je sélectionne un site et que je valide, alors un nouveau website est crée et la modale se ferme", async () => {
      const { store, actions, di } = createStoreForTests()

      const domain = { id: "www.sudoku.academy" }

      di.WebsitesRepository.__storeFetchGoogleDomainsResponse([domain])

      await store.dispatch<any>(actions.websites.$WebsiteCreateModal())

      store.dispatch<any>(
        actions.websites.WebsiteAddSourceModalSelect({ value: domain.id })
      )

      await store.dispatch<any>(actions.websites.$activate())

      expect(store.getState().websites.createWebsiteModal.isOpen).toEqual(false)
      expect(store.getState().websites.entities.length).toEqual(1)
      expect(store.getState().websites.entities[0]).toEqual(domain.id)
      expect(store.getState().websites.map.get(domain.id)).toMatchSnapshot()
    })

    it("en tant qu'utilisateur, si j'essaie d'ajouter un site, sans en avoir sélectionné un au préalable, alors il ne se passe rien", async () => {
      const { store, actions, di } = createStoreForTests()

      const domain = { id: "www.sudoku.academy" }

      di.WebsitesRepository.__storeFetchGoogleDomainsResponse([domain])

      await store.dispatch<any>(actions.websites.$WebsiteCreateModal())

      await store.dispatch<any>(actions.websites.$activate())

      expect(store.getState().websites.createWebsiteModal.isOpen).toEqual(true)
      expect(store.getState().websites.entities.length).toEqual(0)
    })

    it("en tant qu'utilisateur, si l'api renvoie une erreur, afficher une notification", async () => {
      const { store, actions, di } = createStoreForTests()

      const domain = { id: "should_return_error_on_activate" }

      di.WebsitesRepository.__storeFetchGoogleDomainsResponse([domain])

      await store.dispatch<any>(actions.websites.$WebsiteCreateModal())

      store.dispatch<any>(
        actions.websites.WebsiteAddSourceModalSelect({ value: domain.id })
      )

      expect(store.getState().websites.createWebsiteModal.isFetching).toEqual(
        false
      )

      const promise = store.dispatch<any>(actions.websites.$activate())

      expect(store.getState().websites.createWebsiteModal.isFetching).toEqual(
        true
      )

      await promise

      expect(store.getState().websites.createWebsiteModal.isFetching).toEqual(
        false
      )
      expect(store.getState().websites.createWebsiteModal.isOpen).toEqual(true)
      expect(store.getState().websites.entities.length).toEqual(0)
      expect(store.getState().notifications.notifications.length).toEqual(1)
    })
  })

  describe("$fetchAll test suite", () => {
    it("si l'api retourne une erreur, alors afficher une notification", async () => {
      const { store, actions, di } = createStoreForTests()

      di.WebsitesRepository.__setFetchResponse({
        error: true,
        code: ErrorEntity.WEBSITE_NOT_FOUND,
      })

      const promise = store.dispatch<any>(actions.websites.$fetchAll())

      expect(store.getState().websites.fetching).toEqual(true)

      await promise

      expect(store.getState().websites.fetching).toEqual(false)
      expect(store.getState().notifications.notifications.length).toEqual(1)
      expect(store.getState().notifications.notifications[0].message).toEqual(
        ErrorEntity.WEBSITE_NOT_FOUND
      )
    })

    it("je veux pouvoir récupérer mes sites web", async () => {
      const { store, actions, di } = createStoreForTests()

      di.WebsitesRepository.store(WebsiteActivated)

      const promise = store.dispatch<any>(actions.websites.$fetchAll())

      expect(store.getState().websites.fetching).toEqual(true)

      await promise

      expect(store.getState().websites.fetching).toEqual(false)
      expect(store.getState().websites.map).toMatchSnapshot()
    })
  })

  describe.skip("$syncWebsiteAndCheckEverything test suite", () => {
    it("si un utilisateur se rend sur la page indexation de son site, le site doit être mis dans le store et sélectionné", async () => {
      const { store, actions, di } = createStoreForTests()

      di.LocationService.navigate("/indexation/www.sudoku.academy")

      await store.dispatch<any>(
        actions.websites.$syncWebsiteAndCheckEverything()
      )

      expect(store.getState().websites.activeWebsite).toEqual(
        "www.sudoku.academy"
      )
    })
  })

  describe("sitemap flow", () => {
    it("should return an error if I try to submit an empty sitemap", async () => {
      const { store, actions, di } = createStoreForTests()

      const website = WebsiteNoSitemap

      di.WebsitesRepository.__checkResponse({
        website: website.id,
        response: {
          isCredentialsValid: false,
          isSitemapValid: false,
        },
      })

      await store.dispatch<any>(actions.websites.$fetchAll())
      await store.dispatch<any>(
        actions.websites.$changeWebsite({ websiteId: website.id })
      )
      expect(store.getState().websites.addSitemap.isOpen).toEqual(true)

      const sitemap = ""

      store.dispatch<any>(actions.websites.updateSitemap({ value: sitemap }))

      await store.dispatch<any>(actions.websites.$saveSitemap())

      expect(store.getState().websites.addSitemap.isFetching).toEqual(false)
      expect(store.getState().websites.addSitemap.isOpen).toEqual(true)
      expect(store.getState().notifications.notifications[0].message).toEqual(
        NotificationMessageEntity.WEBSITES_SITEMAP_UPDATE_EMPTY
      )
    })

    it("should return an error if the server respond with an error", async () => {
      const { store, actions, di } = createStoreForTests()

      const website = WebsiteNoSitemap

      di.WebsitesRepository.__checkResponse({
        website: website.id,
        response: {
          isCredentialsValid: false,
          isSitemapValid: false,
        },
      })

      di.WebsitesRepository.__updateSitemapResponse({
        website: website.id,
        response: {
          error: true,
          code: ErrorEntity.SITEMAP_INVALID,
        },
      })

      await store.dispatch<any>(actions.websites.$fetchAll())
      await store.dispatch<any>(
        actions.websites.$changeWebsite({ websiteId: website.id })
      )
      expect(store.getState().websites.addSitemap.isOpen).toEqual(true)

      const sitemap = "bad-sitemap.xml"

      store.dispatch<any>(actions.websites.updateSitemap({ value: sitemap }))

      await store.dispatch<any>(actions.websites.$saveSitemap())

      expect(store.getState().websites.addSitemap.isFetching).toEqual(false)
      expect(store.getState().websites.addSitemap.isOpen).toEqual(true)
      expect(store.getState().notifications.notifications[0].message).toEqual(
        ErrorEntity.SITEMAP_INVALID
      )
    })

    it("as a user, i should be able to submit the sitemap", async () => {
      const { store, actions, di } = createStoreForTests()

      const website = WebsiteNoSitemap

      di.WebsitesRepository.store(website)

      di.WebsitesRepository.__checkResponse({
        website: website.id,
        response: {
          isCredentialsValid: false,
          isSitemapValid: false,
        },
      })

      await store.dispatch<any>(actions.websites.$fetchAll())

      await store.dispatch<any>(
        actions.websites.$changeWebsite({ websiteId: website.id })
      )

      expect(store.getState().websites.addSitemap.isOpen).toEqual(true)

      const sitemap = "https://www.sudoku.academy/sitemap.xml"

      store.dispatch<any>(actions.websites.updateSitemap({ value: sitemap }))

      const promise = store.dispatch<any>(actions.websites.$saveSitemap())

      expect(store.getState().websites.addSitemap.isFetching).toEqual(true)

      await promise

      expect(store.getState().websites.map.get(website.id)?.sitemap).toEqual(
        sitemap
      )
      expect(store.getState().websites.addSitemap.value).toEqual(null)
      expect(store.getState().websites.addSitemap.isOpen).toEqual(false)
      expect(store.getState().websites.addSitemap.isFetching).toEqual(false)
    })
  })

  describe("credentials flow", () => {
    it("should be able to update the credentials and close modal if the servers says it's good", async () => {
      const { store, actions, di } = createStoreForTests()

      di.WebsitesRepository.store(WebsiteNoSitemap)
      di.WebsitesRepository.store(WebsiteActivated)

      di.WebsitesRepository.__checkResponse({
        website: WebsiteNoSitemap.id,
        response: {
          isCredentialsValid: false,
          isSitemapValid: false,
        },
      })
      di.WebsitesRepository.__updateCredentialsResponse({
        website: WebsiteNoSitemap.id,
        response: {
          error: false,
          body: null,
        },
      })

      await store.dispatch<any>(actions.websites.$fetchAll())
      await store.dispatch<any>(
        actions.websites.$changeWebsite({ websiteId: WebsiteNoSitemap.id })
      )

      store.dispatch<any>(
        actions.websites.setCredentialsIsOpen({
          value: true,
          website: null,
        })
      )

      expect(store.getState().websites.addCredentials.isOpen).toEqual(true)

      store.dispatch<any>(
        actions.websites.updateCredentials({ value: "the keys" })
      )

      const promise = store.dispatch<any>(actions.websites.$saveCredentials())

      expect(store.getState().websites.addCredentials.isFetching).toEqual(true)

      await promise

      expect(store.getState().websites.addCredentials.isFetching).toEqual(false)
      expect(store.getState().websites.addCredentials.isOpen).toEqual(false)
      expect(store.getState().websites.addCredentials.value).toEqual(null)
    })

    it("should show an error if the update credentials failed", async () => {
      const { store, actions, di } = createStoreForTests()

      di.WebsitesRepository.store(WebsiteNoSitemap)
      di.WebsitesRepository.store(WebsiteActivated)

      di.WebsitesRepository.__checkResponse({
        website: WebsiteNoSitemap.id,
        response: {
          isCredentialsValid: false,
          isSitemapValid: false,
        },
      })
      di.WebsitesRepository.__updateCredentialsResponse({
        website: WebsiteNoSitemap.id,
        response: {
          error: true,
          code: ErrorEntity.GOOGLE_UNKNOWN_ERROR,
        },
      })

      await store.dispatch<any>(actions.websites.$fetchAll())
      await store.dispatch<any>(
        actions.websites.$changeWebsite({ websiteId: WebsiteNoSitemap.id })
      )

      store.dispatch<any>(
        actions.websites.setCredentialsIsOpen({
          value: true,
          website: null,
        })
      )

      store.dispatch<any>(
        actions.websites.updateCredentials({ value: "the keys" })
      )

      await store.dispatch<any>(actions.websites.$saveCredentials())

      expect(store.getState().websites.addCredentials.isFetching).toEqual(false)
      expect(store.getState().websites.addCredentials.isOpen).toEqual(true)
      expect(store.getState().notifications.notifications[0].message).toEqual(
        ErrorEntity.GOOGLE_UNKNOWN_ERROR
      )
    })
  })

  describe("$syncAnalytics test suite", () => {
    it("doit pouvoir récupérer le status et le stocker dans le reducer", async () => {
      const { store, actions, di } = createStoreForTests()

      di.WebsitesRepository.store(WebsiteActivated)

      di.LocationService.navigate(`/analytics/${WebsiteActivated.id}`)

      await store.dispatch<any>(actions.websites.$fetchAll())
      await store.dispatch<any>(
        actions.websites.$changeWebsite({ websiteId: WebsiteActivated.id })
      )

      expect(store.getState().ranking.isFinished).toEqual(true)
    })
  })
})
