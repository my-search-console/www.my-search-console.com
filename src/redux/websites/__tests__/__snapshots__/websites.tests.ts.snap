// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`websites tests suite $fetchAll test suite je veux pouvoir récupérer mes sites web 1`] = `
Map {
  "already_activated:true" => {
    "already_activated": false,
    "bing_domain": null,
    "id": "already_activated:true",
    "image": "null",
    "index_now_installed": false,
    "index_now_key": "1234",
    "is_analytics_activated": true,
    "is_analytics_sync_done": false,
    "is_premium": false,
    "search_console_domain": "sc-domain:already_activated:true",
    "sitemap": "https://already_activated:true",
    "yandex_domain": null,
  },
}
`;

exports[`websites tests suite creation d'un site web en tant qu'utilisateur, si je sélectionne un site et que je valide, alors un nouveau website est crée et la modale se ferme 1`] = `
{
  "already_activated": true,
  "bing_domain": null,
  "id": "www.sudoku.academy",
  "image": "",
  "index_now_installed": false,
  "index_now_key": null,
  "is_analytics_activated": false,
  "is_analytics_sync_done": false,
  "is_premium": false,
  "search_console_domain": "www.sudoku.academy",
  "sitemap": null,
  "yandex_domain": null,
}
`;
